src/qt/pivx/dashboardwidget.cpp

```diff
@@ -182,7 +182,25 @@ void DashboardWidget::loadWalletModel()
         filter->setFilterCaseSensitivity(Qt::CaseInsensitive);
         filter->setSortRole(Qt::EditRole);
         filter->setSourceModel(txModel);
-        filter->sort(TransactionTableModel::Date, Qt::DescendingOrder);
+
+        // Read filter settings
+        QSettings settings;
+        int filterIndex = settings.value("transactionType", 0).toInt(); // Index in ComboBox
+        // Reset transaction type filter when upgrading
+        // TODO: update this if/when we add or remove transaction types
+        if (filterIndex < 0 || filterIndex > 10) filterIndex = 0;
```

Comment:
we can remove this TODO. Instead of using an hardcoded value of 10, just do 
```cpp
if (filterIndex < 0 || filterIndex >= ui->comboBoxSortType->count())
```

Reply:
good call, updated